node/pkg/watchers/aptos/watcher.go

```diff
@@ -128,9 +313,14 @@ func (e *Watcher) Run(ctx context.Context) error {
 
 			logger.Info("Received obsv request", zap.Uint64("tx_hash", nativeSeq))
 
-			s := fmt.Sprintf(`%s?start=%d&limit=1`, eventsEndpoint, nativeSeq)
+			query := fmt.Sprintf(`query GetVersionBySequence { msg(where: {sequence_num: {_eq: %d}}) { version } }`, nativeSeq)
 
-			body, err := e.retrievePayload(s)
+			// Add authorization header for indexer API calls
+			headers := make(map[string]string)
```

Comment:
Nit: this field is static so it could be defined once outside of the ticker loop